src/main/java/hudson/remoting/ClassFilter.java

```diff
@@ -187,21 +199,27 @@ public static void appendDefaultFilter(Pattern filter) {
 
         private final List<Object> blacklistPatterns;
 
-        RegExpClassFilter(String[] patterns) throws PatternSyntaxException {
+        RegExpClassFilter(String[] patterns) throws ClassFilterException {
             blacklistPatterns = new ArrayList<>(patterns.length);
             for (String pattern : patterns) {
                 add(pattern);
             }
         }
 
-        void add(String pattern) throws PatternSyntaxException {
+        void add(String pattern) throws ClassFilterException {
+
             if (OPTIMIZE1.matcher(pattern).matches()) {
                 // this is a simple startsWith test, no need to slow things down with a regex
                 blacklistPatterns.add(pattern.substring(1, pattern.length() - 2).replace("[.]", "."));
             } else if (OPTIMIZE2.matcher(pattern).matches()) {
                 // this is a simple startsWith test, no need to slow things down with a regex
                 blacklistPatterns.add(pattern.substring(3, pattern.length() - 4));
             } else {
+                try {
+                    Pattern regex = Pattern.compile(pattern);
+                } catch (PatternSyntaxException ex) {
```

Comment:
Impossible, because we are either in `createDefaultInstance`, for which the patterns are statically defined and known to be valid; or in `appendDefaultFilter`, which takes a `Pattern` which is by definition well-formed.